### PR TITLE
create and use `svg_icon` helper which defines icon if needed and then uses it.

### PR DESCRIPTION
**Pull request by timdiggins**
Issues I've just thought of:

* will this mess with caching (definition should probably be hoiked outside of cachable content)
* is "svg_icon" the right name for this.

### CODE REVIEW

File: app/helpers/thredded/icon_helper.rb [hunk 1]
```diff
@@ -2,6 +2,17 @@
 
 module Thredded
   module IconHelper
+    def svg_icon(filename, **args)
```
Comment (glebm) [inline app/helpers/thredded/icon_helper.rb]: `shared_inline_svg`?

**Comment by glebm** [comment]
> will this mess with caching (definition should probably be hoiked outside of cachable content)

It would, but we don't cache anything except post contents and I don't think we will.

File: app/helpers/thredded/icon_helper.rb (hunk 1 above)
Comment (timdiggins) [reply app/helpers/thredded/icon_helper.rb]: Yes good. I’ll do that then merge this into the other branch

Comment (timdiggins) [reply app/helpers/thredded/icon_helper.rb]: Yes good. I’ll do that then merge this into the other branch

**Comment by jayroh** [comment]
There are several rails helper gems that provide a helper like this I believe. That's not to say we shouldn't or couldn't have our own - but to warn against any possible clashes between method names.

I'm in my phone currently so I'm not able to do a lot of digging but a look through Ruby toolbox for the more prevailantly used gems might be a good guard against that

**Comment by timdiggins** [comment]
@jayroh I've looked at https://www.ruby-toolbox.com/search?utf8=%E2%9C%93&q=svg and the only thing in the same space I can find is inline_svg, which this is wrapping

Also, FYI I looked into our svgs inline but in a separate (browser-cachable) document https://css-tricks.com/svg-use-with-external-reference-take-2/. This helper puts us on the way to doing that, but unfortunately because svg use doesn't support CORS, the document with the svgs in it would have to be served by the same host as the web page that "uses" them  -- which would exempt CDNs, and might thus cause more problems than it solves. (but i still could imagine this working -- but we'd want to link to a compiled (and digested) asset (probably an html file), but using the current host rather than the asset_url. I don't know whether that's possible, but in any case out of scope for the thredded project per se).

**Comment by glebm** [comment]
> Also, FYI I looked into our svgs inline but in a separate (browser-cachable) document

Another issue is that it would cause flash of unstyled content (missing SVGs) on the first load (IMO for no good reason given how tiny these SVGs are).

**Pull request merged**